scipy/optimize/_minpack_py.py

```diff
@@ -870,6 +887,25 @@ def curve_fit(f, xdata, ydata, p0=None, sigma=None, absolute_sigma=False,
     if ydata.size == 0:
         raise ValueError("`ydata` must not be empty!")
 
+    # nan handling is needed only if check_finite is False because if True,
+    # the x-y data are already checked, and they don't contain nans.
+    if not check_finite and nan_policy is not None:
+        if nan_policy == "propagate":
+            raise ValueError("`nan_policy='propagate'` is not supported "
+                             "by this function.")
+
+        x_contains_nan, nan_policy = _contains_nan(xdata, nan_policy)
```

Comment:
Here's another rough edge.

If `nan_policy='hi'`, `_contains_nan` produces:
```
ValueError: nan_policy must be one of {'propagate', 'raise', 'omit'}
```
but that's not accurate here.

I think the fix is to make it:
```
ValueError: nan_policy must be one of {'propagate', 'raise', 'omit', None}
```
but then I need to adjust `_axis_nan_policy` stuff in stats to accept `nan_policy=None`.

Reply:
Resolved by gh-17852.